nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-shell-authorizer/src/test/java/org/apache/nifi/authorization/ShellUserGroupProviderIT.java

```diff
@@ -0,0 +1,215 @@
+
+import org.apache.nifi.authorization.util.ShellRunner;
+
+
+public class ShellUserGroupProviderIT extends ShellUserGroupProviderBase {
+    private static final Logger logger = LoggerFactory.getLogger(ShellUserGroupProviderIT.class);
+
+    private final static String ALPINE_IMAGE = "natural/alpine-sshd:latest";
+    private final static String CENTOS_IMAGE = "natural/centos-sshd:latest";
+    private final static String DEBIAN_IMAGE = "natural/debian-sshd:latest";
+    private final static String UBUNTU_IMAGE = "natural/ubuntu-sshd:latest";
+    private final static List<String> TEST_CONTAINER_IMAGES =
+        Arrays.asList(
+                      ALPINE_IMAGE,
+                      CENTOS_IMAGE,
+                      DEBIAN_IMAGE,
+                      UBUNTU_IMAGE
+                      );
+
+    private final static String CONTAINER_SSH_AUTH_KEYS = "/root/.ssh/authorized_keys";
+    private final static Integer CONTAINER_SSH_PORT = 22;
+
+    private static String sshPrivKeyFile;
+    private static String sshPubKeyFile;
+
+    private AuthorizerConfigurationContext authContext;
+    private ShellUserGroupProvider localProvider;
+    private UserGroupProviderInitializationContext initContext;
+
+    @ClassRule
+    static public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Before
+    public void setup() throws IOException {
+        authContext = mock(AuthorizerConfigurationContext.class);
+        initContext = mock(UserGroupProviderInitializationContext.class);
+
+        localProvider = new ShellUserGroupProvider();
+        try {
+            localProvider.initialize(initContext);
+            localProvider.onConfigured(authContext);
+        } catch (final Exception exc) {
+            systemCheckFailed = true;
+            logger.error("setup() exception: " + exc + "; tests cannot run on this system.");
+            return;
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() throws IOException {
+        sshPrivKeyFile = tempFolder.getRoot().getAbsolutePath() + "/id_rsa";
+        sshPubKeyFile = sshPrivKeyFile + ".pub";
+
+        try {
+            // NB: this command is a bit perplexing: it works without prompt from the shell, but hangs
+            // here without the pipe from `yes`:
+            ShellRunner.runShell("yes | ssh-keygen -C '' -N '' -t rsa -f " + sshPrivKeyFile);
+        } catch (final IOException ioexc) {
+            systemCheckFailed = true;
+            logger.error("setupOnce() exception: " + ioexc + "; tests cannot run on this system.");
+            return;
+        }
+
+        // Fix the file permissions to abide by the ssh client
+        // requirements:
+        Arrays.asList(sshPrivKeyFile, sshPubKeyFile).forEach(name -> {
+                final File f = new File(name);
+                assertTrue(f.setReadable(false, false));
+                assertTrue(f.setReadable(true));
+            });
+    }
+
+    @Test
+    public void testGetUsers() {
```

Comment:
Is there any expectation that other classes will extend `ShellUserGroupProviderBase`? I understand putting reusable functionality in a common location, but it makes these tests very opaque.

Reply:
Not at this point, no.  The approach seemed best to accommodate the remote commands (ssh to test containers).  I can re-work the test classes if it's a pressing issue.

Reply:
Ok, it doesn't need to be refactored, but the test method names & Javadoc can be made more expressive so the "what" of what each method is asserting is clear to the caller.

Reply:
Renamed the test methods to be more descriptive, and added a small javadoc to each.